src/path.cr

```diff
@@ -653,6 +653,7 @@ struct Path
       parts.each_with_index do |part, index|
         case part
         when Path
+          part = part.to_posix if posix? && part.windows?
```

Comment:
Isn't this missing the other case? if `self.windows?` and `part.posix?`

Reply:
Every POSIX path is also a valid windows path, there is no transformation needed (see `#to_windows`). I should probably add a comment about this.

Reply:
The only issue is, should backslashes be escaped when converting to windows? But this also applies to `#to_windows`.

```cr
Path.posix("foo\\bar").to_windows # => "foo/bar" or "foo\\bar"?
```

I guess it would probably better to escape, otherwise the semantics would change between posix and windows. Conversions should be reversable, so `Path.posix("foo\\bar").to_windows.to_posix == Path.posix("foo\\bar")` is true.

Reply:
Shouldn't that raise, there's no way to represent the `\` character in a filename on windows.